### PR TITLE
tests: reboot after upgrading to snapd on the -proposed pocket

### PR DESCRIPTION
**Pull request by fgimenez**
In the automated SRU validations for trusty we were getting errors because the changes after installing snapd from the `-proposed` pocket were not in place and a reboot was required (mostly because of kernel upgrades). With these changes that reboot takes place and we have a behavior similar to autopkgtest.

### CODE REVIEW

**Comment by fgimenez** [comment]
@niemeyer the test suite was failing on a (probably outdated) trusty image, snapd from -proposed pulls the latest kernel packages and a reboot is required for the changes to take effect, if not there are errors related to snap-confine. I've tried to check if the reboot is needed looking for /var/run/reboot-required but didn't work (the file is never present), maybe there's a better way, as always suggestions are more than welcome :)

This is how the these SRU validation executions are triggered:
```
export SPREAD_MODIFY_CORE_SNAP_FOR_REEXEC=0
export SPREAD_TRUST_TEST_KEYS=false
export SPREAD_SRU_VALIDATION=1
export SPREAD_SNAP_REEXEC=0
export SPREAD_CORE_CHANNEL=beta

spread -v linode:ubuntu-14.04-64:tests/main/
```

File: tests/lib/apt.sh [hunk 1]
```diff
@@ -11,6 +11,9 @@ install_build_snapd(){
         apt install -y --only-upgrade snapd
         mv sources.list.back /etc/apt/sources.list
         apt update
+        if [ "$SPREAD_REBOOT" = 0 ]; then
```
Comment (sergiocazzolato) [inline tests/lib/apt.sh]: Suggestion: perhaps this could be written like:
[ "$SPREAD_REBOOT" = 0 ] && REBOOT

Comment (fgimenez) [inline tests/lib/apt.sh]: We use the `if` format in the whole suite, I personally find it much more readable. Additionally, it allows us to use echo to let know to anyone trying to debug the test what have you just checked (not in this case though, in which the thing being checked is clear enough).

**Comment by fgimenez** [comment]
@mvo5 thx! very good points, done

**Pull request merged**